src/Core/Form/IdentifiableObject/CommandBuilder/Product/StockAvailableCommandsBuilder.php

```diff
@@ -0,0 +1,103 @@
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Command\UpdateStockCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\CommandBuilder;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\CommandBuilderConfig;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\DataField;
+
+/**
+ * Builds following command for single and all shops:
+ *
+ * @see UpdateStockCommand
+ *
+ * @todo: may be renamed to StockCommandsBuilder when UpdateProductCommand is fully implemented and previous UpdateStockCommandsBuilder is deleted.
+ */
+class StockAvailableCommandsBuilder
+{
+    /**
+     * @var string
+     */
+    private $modifyAllNamePrefix;
+
+    /**
+     * @param string $modifyAllNamePrefix
+     */
+    public function __construct(string $modifyAllNamePrefix)
+    {
+        $this->modifyAllNamePrefix = $modifyAllNamePrefix;
+    }
+
+    public function buildCommands(ProductId $productId, array $formData, ShopConstraint $singleShopConstraint): array
+    {
+        if (!isset($formData['stock']) && !isset($formData['combinations'])) {
+            return [];
+        }
+
+        $productType = $formData['header']['type'] ?? ProductType::TYPE_STANDARD;
+        if ($productType === ProductType::TYPE_COMBINATIONS) {
+            return $this->buildCommandsForProductWithCombinations($productId, $formData, $singleShopConstraint);
+        }
+
+        return $this->buildCommandsForRegularProduct($productId, $formData, $singleShopConstraint);
+    }
+
+    private function buildCommandsForRegularProduct(ProductId $productId, array $formData, ShopConstraint $singleShopConstraint): array
+    {
+        $config = new CommandBuilderConfig($this->modifyAllNamePrefix);
+        $config
+            ->addMultiShopField('[stock][quantities][delta_quantity][delta]', 'setDeltaQuantity', DataField::TYPE_INT)
+            ->addMultiShopField('[stock][options][stock_location]', 'setLocation', DataField::TYPE_STRING)
+            ->addMultiShopField('[stock][availability][out_of_stock_type]', 'setOutOfStockType', DataField::TYPE_INT)
+        ;
+
+        $commandBuilder = new CommandBuilder($config);
+        $shopCommand = new UpdateStockCommand($productId->getValue(), $singleShopConstraint);
+        $allShopsCommand = new UpdateStockCommand($productId->getValue(), ShopConstraint::allShops());
+
+        return $commandBuilder->buildCommands($formData, $shopCommand, $allShopsCommand);
+    }
+
+    private function buildCommandsForProductWithCombinations(ProductId $productId, array $formData, ShopConstraint $singleShopConstraint): array
```

Comment:
It was probably not present in the initial builder but it deserves a comment here that I forgot `For product with combinations we only handle one field out_of_stock_type which is common to all combinations, the delta stock and location are handled combination by combination in another dedicated command`